### ISSUE TITLE
Compiler warnings on Windows 2022 & msys2

### ISSUE THREAD

**Issue by skirpichev** [aleaxit/gmpy#398]
```
  In file included from /usr/include/python3.11/unicodeobject.h:58,
                   from /usr/include/python3.11/Python.h:51,
                   from src/gmpy2.c:505:
  src/gmpy2_convert_gmp.c: In function ‘GMPy_MPQ_From_PyStr’:
  src/gmpy2_convert_gmp.c:648:29: warning: array subscript has type ‘char’ [-Wchar-subscripts]
    648 |                 if (isdigit(*counter))
        |                             ^~~~~~~~
  src/gmpy2_convert_mpc.c: In function ‘GMPy_MPC_From_PyStr’:
  src/gmpy2_convert_mpc.c:295:20: warning: array subscript has type ‘char’ [-Wchar-subscripts]
    295 |     while (isspace(*lastchar))
        |                    ^~~~~~~~~
  src/gmpy2_convert_mpc.c:309:20: warning: array subscript has type ‘char’ [-Wchar-subscripts]
    309 |     while (isspace(*cp))
        |                    ^~~
  src/gmpy2_format.c: In function ‘GMPy_MPZ_Format’:
  src/gmpy2_format.c:131:21: warning: array subscript has type ‘char’ [-Wchar-subscripts]
    131 |         if (isdigit(*p1)) {
        |                     ^~~
  src/gmpy2_format.c: In function ‘GMPy_MPFR_Format’:
  src/gmpy2_format.c:266:21: warning: array subscript has type ‘char’ [-Wchar-subscripts]
    266 |         if (isdigit(*p1)) {
        |                     ^~~
  src/gmpy2_format.c: In function ‘GMPy_MPC_Format’:
  src/gmpy2_format.c:447:36: warning: array subscript has type ‘char’ [-Wchar-subscripts]
    447 |                     while (isdigit(*(ifmtptr-1)))
        |                                    ^~~~~~~~~~~~
  src/gmpy2_format.c:453:21: warning: array subscript has type ‘char’ [-Wchar-subscripts]
    453 |         if (isdigit(*p)) {
        |                     ^~
```
See [this build](https://github.com/aleaxit/gmpy/actions/runs/4401325070/jobs/7707426250).

**Comment by casevh** [comment]
Just an FYI - I've successfully compiled GMP, MPFR, MPC, and gmpy2 using a version of Mingw-64 configured to UCRT. No warnings. I'll keep digging.

**Comment by skirpichev** [comment]
@casevh , I think this issue shouldn't stop https://github.com/aleaxit/gmpy/pull/397 anyway.

**Comment by casevh** [comment]
Agree. Merged #397
